tasks/DependencyCheck/task/index.ts

```diff
@@ -17,65 +20,73 @@ async function run(): Promise<void> {
 
-    await owaspCheck(scriptFullPath, scanPath, excludedScanPathPatterns, csvFilePath, enableSelfHostedDatabase);
-
-    const payload = await csv()
-      .fromFile(csvFilePath)
-      .subscribe((jsonObj: any) => {
-        return new Promise((resolve, reject) => {
-          jsonObj.RepositoryName = repositoryName;
-          jsonObj.BranchName = branchName;
-          jsonObj.BuildName = buildName;
-          jsonObj.BuildNumber = buildNumber;
-          jsonObj.CommitId = commitId;
-          resolve();
+      const la: ILogAnalyticsClient = new LogAnalyticsClient(
+        workspaceId,
+        sharedKey,
+      );
+
+      if (enableSelfHostedVulnerabilityFiles) {
+        await downloadVulnData(readStorageAccountContainerSasUri, `${__dirname}/dependency-check-cli/data/odc.mv.db`, taskVersion);
+        await downloadVulnData(readStorageAccountContainerSasUri, `${__dirname}/dependency-check-cli/data/jsrepository.json`, taskVersion);
+      }
+
+      await owaspCheck(scriptFullPath, scanPath, excludedScanPathPatterns, csvFilePath, enableSelfHostedVulnerabilityFiles);
+
+      const payload = await csv()
+        .fromFile(csvFilePath)
+        .subscribe((jsonObj: any) => {
+          return new Promise((resolve, reject) => {
+            jsonObj.RepositoryName = repositoryName;
+            jsonObj.BranchName = branchName;
+            jsonObj.BuildName = buildName;
+            jsonObj.BuildNumber = buildNumber;
+            jsonObj.CommitId = commitId;
+            resolve();
+          })
         })
-      })
-
-    if (payload.length > 0) {
-      await la.sendLogAnalyticsData(
-        JSON.stringify(payload), logType, new Date().toISOString(),
-      ).then((() => {
-        const vuln = payload.length > 1 ? 'vulnerabilities' : 'vulnerability';
-        tl.warning(`${emoji.get('pensive')}  A total of ${payload.length} ${vuln} were found in this project.`);
-      })).catch(((e) => {
-        tl.setResult(tl.TaskResult.Failed, e);
-      }));
-    } else {
-      console.log(`${emoji.get('slightly_smiling_face')}  Good news, there are no vulnerabilities to report!`);
+
+      if (payload.length > 0) {
+        await la.sendLogAnalyticsData(
+          JSON.stringify(payload), logType, new Date().toISOString(),
+        ).then((() => {
+          const vuln = payload.length > 1 ? 'vulnerabilities' : 'vulnerability';
+          tl.warning(`${emoji.get('pensive')}  A total of ${payload.length} ${vuln} were found in this project.`);
+        })).catch(((e) => {
+          tl.setResult(tl.TaskResult.Failed, e);
+        }));
+      } else {
+        console.log(`${emoji.get('slightly_smiling_face')}  Good news, there are no vulnerabilities to report!`);
+      }
+    }
+    else {
+      await owaspCheck(scriptFullPath, scriptFullPath, excludedScanPathPatterns, csvFilePath, false);
+      await UploadVulnData(writeStorageAccountContainerSasUri, `${__dirname}/dependency-check-cli/data/odc.mv.db`, taskVersion);
```

Comment:
casing on UploadVulnData is inconsistent, other function names are camal cased

Reply:
Updated to camelCase